Automattic/jetpack
# Add upgrade CTAs to the Site Stats page

This is the first PR of the project to add upgrade CTAs to the Site Stats page. It introduces a new hook and a class to handle the job, all the methods to check whether the features are enabled or not and a draft UI.

Everything is hidden behind the `JETPACK_DEV_TEST_STATS_UPGRADE_NUDGES` so we can merge and test it while still developing it. Once everything is done, this constant can be dropped.

#### Changes proposed in this Pull Request:

* Introduces the first iteration of the Upsells in the Site Stats page, still as a work in progress

#### Jetpack product discussion


p1HpG7-cfa-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






What we are reviewing?

In this PR, we are only reviewing the basic structure of the project and the rules we are using in order to dislpay or not the CTAs

What we are NOT reviewing?

We are not reviewing the general appearance of the new section added to the page. HTML, css, icons and copy used are all temporary and will be addressed in future PRs.

* Checkout this PR and connect Jetpack
* Edit your `wp-config.php` and add `define( 'JETPACK_DEV_TEST_STATS_UPGRADE_NUDGES', true );`
* Visit Jetpack > Site Stats
* You should see a new section with the upsells in the bottom
* Check p6TEKc-56L-p2, and you will see all the existing CTAs.
* Buy and activate different features in order to test all possible scenarios and see if only the appropriate CTAs show up. For example, Search should not should up if you have Search enabled.

Code Review
* Does the code organization look good?
* Do the way we are checking for features look good? Some of them, like Backup and Scan, actually rely on remote requests to the WPCOM API, which can be expensive. (but those requests are cached and if the user visited the Jetpack Dashboard it should be cached already...) is there a better way?
* I added the section outside of the dynamic content of the page, that is loaded async. Do you think it should be inside?
